### PR TITLE
Document branches in contributing guidelines

### LINKED ISSUE #1232: Contributing guidelines readme overlaps with contributing
There is an overlap in DCO, sign your work and license sections.

The purpose of `contributing_guidelines.md` is to have a generic readme file which applies to the whole org. Oe this readme could reference it https://github.com/jaegertracing/jaeger-operator/blob/master/CONTRIBUTING.adoc#certificate-of-origin

### PR DESCRIPTION
**Pull request by pavolloffay**
Also resolves #1232

### CODE REVIEW

File: CONTRIBUTING_GUIDELINES.md [hunk 1]
```diff
@@ -91,6 +92,13 @@ message:
 using your real name (sorry, no pseudonyms or anonymous contributions.) You can
 add the sign off when creating the git commit via `git commit -s`.
 
+If you want signing to be automatic you can set up some aliases:
+
+```
+git config --add alias.amend "commit -s --amend"
+git config --add alias.c "commit -s"
```
Comment (yurishkuro) [inline CONTRIBUTING_GUIDELINES.md]: Do these actually work? I tried once but it didn't help.

Comment (pavolloffay) [reply CONTRIBUTING_GUIDELINES.md]: I will try, if they don't work I will remove them.

Comment (pavolloffay) [reply CONTRIBUTING_GUIDELINES.md]: Does not work for me I will remove it

**Comment by pavolloffay** [comment]
@yurishkuro @vprithvi @black-adder  @tiffon please do not push feature branches to upstream and rather submit PR from your fork.

https://github.com/jaegertracing/jaeger/branches/all

**Comment by yurishkuro** [comment]
I have already changed my git clone.

File: CONTRIBUTING_GUIDELINES.md [hunk 2]
```diff
@@ -147,3 +150,8 @@ git push --force
 
 [good-commit-msg]: https://chris.beams.io/posts/git-commit/
 [dco-bot]: https://github.com/probot/dco#how-it-works
+
+## Branches
+
+Upstream repository should contain only maintenance branches (e.g. `release-1.0`). For feature
+branches use forked repository.
```
Comment (yurishkuro) [inline CONTRIBUTING_GUIDELINES.md]: We may want to add a bit of explanation here. In particular, one critical step that does not happen automatically with a fork is to make sure the local `master` branch tracks the upstream, not your fork (origin).
```shell
$ git remote -v
origin	git@github.com:jaegertracing/jaeger.git (fetch)
origin	git@github.com:jaegertracing/jaeger.git (push)

$ git remote rm origin

$ git remote add upstream git@github.com:jaegertracing/jaeger.git

$ git remote add origin git@github.com:yurishkuro/jaeger.git

$ git fetch upstream master
remote: Enumerating objects: 13, done.
remote: Counting objects: 100% (13/13), done.
remote: Total 19 (delta 12), reused 13 (delta 12), pack-reused 6
Unpacking objects: 100% (19/19), done.
From github.com:jaegertracing/jaeger
 * branch            master     -> FETCH_HEAD
 * [new branch]      master     -> upstream/master

$ git pull
There is no tracking information for the current branch.
Please specify which branch you want to merge with.
See git-pull(1) for details.

    git pull <remote> <branch>

If you wish to set tracking information for this branch you can do so with:

    git branch --set-upstream-to=<remote>/<branch> master

$ git branch --set-upstream-to=upstream/master master
Branch 'master' set up to track remote branch 'master' from 'upstream'.

$ git remote -v
origin    git@github.com:yurishkuro/jaeger.git (fetch)
origin    git@github.com:yurishkuro/jaeger.git (push)
upstream  git@github.com:jaegertracing/jaeger.git (fetch)
upstream  git@github.com:jaegertracing/jaeger.git (push)
```

Without this, the local master tracks for fork and you need to do extra steps to merge upstream master into it. People often forget to do that and end up with PRs having all kinds of weird commits in them.

**Pull request merged**